src/coreclr/src/jit/compiler.cpp

```diff
@@ -4395,94 +4407,84 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
-    }
-#endif // DEBUG
+        // Initialize the BlockSet epoch
+        NewBasicBlockEpoch();
 
-    // Insert call to class constructor as the first basic block if
-    // we were asked to do so.
-    if (info.compCompHnd->initClass(nullptr /* field */, info.compMethodHnd /* method */,
-                                    impTokenLookupContextHandle /* context */) &
-        CORINFO_INITCLASS_USE_HELPER)
-    {
-        fgEnsureFirstBBisScratch();
-        fgNewStmtAtBeg(fgFirstBB, fgInitThisClass());
-    }
+        fgOutgoingArgTemps = nullptr;
+
+        // Insert call to class constructor as the first basic block if
+        // we were asked to do so.
+        if (info.compCompHnd->initClass(nullptr /* field */, info.compMethodHnd /* method */,
+                                        impTokenLookupContextHandle /* context */) &
+            CORINFO_INITCLASS_USE_HELPER)
+        {
+            fgEnsureFirstBBisScratch();
+            fgNewStmtAtBeg(fgFirstBB, fgInitThisClass());
+        }
 
 #ifdef DEBUG
-    if (opts.compGcChecks)
-    {
-        for (unsigned i = 0; i < info.compArgsCount; i++)
+        if (opts.compGcChecks)
         {
-            if (lvaTable[i].TypeGet() == TYP_REF)
+            for (unsigned i = 0; i < info.compArgsCount; i++)
             {
-                // confirm that the argument is a GC pointer (for debugging (GC stress))
-                GenTree*          op   = gtNewLclvNode(i, TYP_REF);
-                GenTreeCall::Use* args = gtNewCallArgs(op);
-                op                     = gtNewHelperCallNode(CORINFO_HELP_CHECK_OBJ, TYP_VOID, args);
+                if (lvaTable[i].TypeGet() == TYP_REF)
+                {
+                    // confirm that the argument is a GC pointer (for debugging (GC stress))
+                    GenTree*          op   = gtNewLclvNode(i, TYP_REF);
+                    GenTreeCall::Use* args = gtNewCallArgs(op);
+                    op                     = gtNewHelperCallNode(CORINFO_HELP_CHECK_OBJ, TYP_VOID, args);
 
-                fgEnsureFirstBBisScratch();
-                fgNewStmtAtEnd(fgFirstBB, op);
+                    fgEnsureFirstBBisScratch();
+                    fgNewStmtAtEnd(fgFirstBB, op);
 
-                if (verbose)
-                {
-                    printf("\ncompGcChecks tree:\n");
-                    gtDispTree(op);
+                    if (verbose)
+                    {
+                        printf("\ncompGcChecks tree:\n");
+                        gtDispTree(op);
+                    }
                 }
             }
         }
-    }
 #endif // DEBUG
 
-#if defined(DEBUG) && defined(TARGET_XARCH)
-    if (opts.compStackCheckOnRet)
-    {
-        lvaReturnSpCheck                  = lvaGrabTempWithImplicitUse(false DEBUGARG("ReturnSpCheck"));
-        lvaTable[lvaReturnSpCheck].lvType = TYP_I_IMPL;
-    }
-#endif // defined(DEBUG) && defined(TARGET_XARCH)
+#if defined(DEBUG) && defined(_TARGET_XARCH_)
```

Comment:
Didn't you convert these #ifdefs to the old form? Shouldn't you leave them without leading/trailing underscores?

Reply:
Thanks for spotting that.

This PR is a revival of some older work, so it has the old ifdef names. Need to keep reminding myself to watch for them.